core/server/data/sitemap/base-generator.js

```diff
@@ -24,10 +25,16 @@ function BaseSiteMapGenerator() {
 }
 
 _.extend(BaseSiteMapGenerator.prototype, {
+    dataEvents: events,
```

Comment:
I think you'll want this as an instance value in the constructor (and maybe allow someone to pass it in (like for unit tests) but default to events), or others could potentially overwrite for all instances.